src/Sulu/Component/SmartContent/ContentType.php

```diff
@@ -285,13 +284,9 @@ public function getContentData(PropertyInterface $property)
                 $params,
                 $options,
                 (!empty($limit) ? \intval($limit) : null),
-                $page,
+                $page < 1 ? 1 : ($page > \PHP_INT_MAX ? \PHP_INT_MAX : $page),
```

Comment:
why this change? is `page` a string here? would a cast to int not do the same? I think I would do a return (int)  in the getCurrentPage method.

Reply:
In the tests we expect `$page` to be a valid integer > 0 and < `PHP_INT_MAX`. Before any `$page` outside these boundaries would have thrown the exception and thus I thought it would also make sense here to show the first page instead of trying to show a negative page value. A negative page value results in a doctrine exception `Offset must be a positive integer or zero`

Reply:
Would still do the logic in the `getCurrentPage` method then.

Reply:
Yeah makes more sense, I think we could also omit the `$page > PHP_INT_MAX` check and only check for negative integers.

Reply:
At the end we read a query parameter here and a query parameter will always be a string representation. 

Its just funny that there is a difference in PHP converting a float to int and converting a string to int:

```bash
php -r 'var_dump((int) 9223372036854775808); var_dump((int) "9223372036854775807");';

int(-9223372036854775808)
int(9223372036854775807)
```

So from real world usage the first one can not appear (as query strings are always strings).

I would go in the getCurrentPage with:

```php
if ($page <= 1) {
     return 1;
}

if ( $page > \PHP_INT_MAX) {
      return \PHP_INT_MAX;
}

return (int) $page;
```

Reply:
I would also adopt the tests if **easy possible** that the `$request->query->getParameter(` will return a string value not a float or int. As this matches the real world usage.